code/client/installhelper

```diff
@@ -0,0 +1,376 @@
+    launchd_labels = split_launchctl_list(cmd_out)
+
+    # For each plist in /Library/LaunchDaemons/ prefixed with LAUNCHD_PREFIX
+    for launch_daemon in launch_daemons:
+        try:
+            # Make sure content is valid, before proceeding to reload
+            daemon_content = FoundationPlist.readPlist(launch_daemon)
+        except FoundationPlist.NSPropertyListSerializationException:
+            break
+        # Get the Label
+        daemon_label = daemon_content.get('Label')
+        # If we have a label
+        if daemon_label:
+            # Check to see if the launch daemon is loaded, before stopping
+            if daemon_label in launchd_labels:
+                # Stop the launch daemon at the login window
+                log("Stopping: %s" % launch_daemon)
+                cmd = ['/bin/launchctl', 'bootout', 'system/' + daemon_label]
+                subprocess.call(cmd)
+            # Load the launch daemon at the login window
+            log("Starting: %s" % launch_daemon)
+            cmd = ['/bin/launchctl', 'bootstrap', 'system', launch_daemon]
+            subprocess.call(cmd)
+
+
+def reload_loginwindow_launch_agents(launch_agents):
+    '''Reloads Munki's loginwindow Launch Agents'''
+
+    # Log what we're upto
+    log("Reloading LoginWindow Launch Agents")
+
+    # Retrieve all running launch agents at the login window
+    cmd = ['/bin/launchctl', 'list']
+    cmd_out = subprocess.run(cmd, capture_output = True, check = False)
+    launchd_labels = split_launchctl_list(cmd_out)
+
+    # For each plist in /Library/LaunchAgents/ prefixed with LAUNCHD_PREFIX
+    for launch_agent in launch_agents:
+        try:
+            # Make sure content is valid, before proceeding to reload
+            agent_content = FoundationPlist.readPlist(launch_agent)
+        except FoundationPlist.NSPropertyListSerializationException:
+            break
+        # If we're to limit loading
+        if agent_content.get('LimitLoadToSessionType'):
+            # If the launch agent includes LimitLoadToSessionType, and it contains LoginWindow
+            if 'LoginWindow' in agent_content.get('LimitLoadToSessionType'):
+                # If the launch agent is to be loaded at the login window, get the Label
+                agent_label = agent_content.get('Label')
+                if agent_label:
+                    # Check to see if the launch agent is loaded, before stopping
+                    if agent_label in launchd_labels:
+                        # Stop the launch agent at the login window
+                        log("Stopping: %s" % launch_agent)
+                        cmd = ['/bin/launchctl', 'bootout', 'loginwindow/' + agent_label]
+                        subprocess.call(cmd)
+                    # Load the launch agent at the login window
+                    log("Starting: %s" % launch_agent)
+                    cmd = ['/bin/launchctl', 'bootstrap', 'loginwindow', launch_agent]
```

Comment:
```
/bin/launchctl load -S loginwindow /Library/LaunchAgents/com.googlecode.munki.MunkiStatus.plist 
Load failed: 5: Input/output error
Try running `launchctl bootstrap` as root for richer errors.
```

And as root:
```
/bin/launchctl load -S loginwindow /Library/LaunchAgents/com.googlecode.munki.MunkiStatus.plist
Warning: Expecting a LaunchDaemons path since the command was ran as root. Got LaunchAgents instead.
`launchctl bootstrap` is a recommended alternative.
Load failed: 5: Input/output error
Try running `launchctl bootstrap` as root for richer errors.
```

I'll need to double check to make sure things are loaded and just not an error we can ignore.

Reply:
Sorry confused the capitalisation.. loginwindow is the v2.0 service.. LoginWindow for v1.0

Reply:
Resolved in: https://github.com/munki/munki/pull/1203/commits/c1bae70d6f8cec748aab947ba252f8d2b33c5108

Reply:
Warnings aren't necessarily errors, but man that's annoying. `sudo launchctl load -S LoginWindow /Library/LaunchAgents/com.googlecode.munki.MunkiStatus.plist` should be a valid command. And if `sudo launchctl bootstrap loginwindow /Library/LaunchAgents/com.googlecode.munki.MunkiStatus.plist` is a valid command, then _Apple should document it_.

But I rather suspect they just want loginwindow LaunchAgents to go away, so they're hoping that if they ignore them and make the tooling confusing we'll just stop using them :-(

Reply:
The errors where my fault in using loginwindow and not LoginWindow.. corrected in the last commit.